File: Hritwik/Problems/Milestone1/Problem28.py
class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right
class Solution:
    def increasingBST(self, root: TreeNode) -> TreeNode:
        response = []
        def inorder_(node):
            if node is None:
                return
            if node.left:
                inorder_(node.left)
            response.append(node.val)
            if node.right:
                inorder_(node.right)
        inorder_(root)
        response_head=TreeNode(-1,None)
        response_iter = response_head
        for node in response:
            tree_node = TreeNode(node, None)
            response_iter.right = tree_node
            response_iter = response_iter.right
        return response_head.right
    
# Create Original binary tree to traverse through it in an inorder traversal manner
#
#     Original    4
#              /     \
#             3        5  
#            / \        \
#           1   2        7

original = TreeNode(4)
original.left = TreeNode(3)
original.left.left = TreeNode(1)
original.left.right = TreeNode(2)
original.right = TreeNode(5)
original.right.right = TreeNode(7)

solution = Solution()
response = solution.increasingBST(original)
print(f'Response root.: {response}')

File: SBiswas/Problems/Milestone1/Problem28.cpp
#include<iostream>
#include<vector>
#include<unordered_map>

using namespace std;


//  Definition for a binary tree node.
struct TreeNode {
    int val;
    TreeNode *left;
    TreeNode *right;
    TreeNode() : val(0), left(nullptr), right(nullptr) {}
    TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
    TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 };

class Solution {

    TreeNode* temp = new TreeNode;
    TreeNode* ans = temp;
    TreeNode* last_val_node = nullptr;
public:

    void inorder(TreeNode* root)
    {
        if(!root)
        {
            return;
        }

        inorder(root->left);
        last_val_node = temp;
        temp->val = root->val;
        temp->right = new TreeNode;
        temp = temp->right;
        inorder(root->right);
    }

    TreeNode* increasingBST(TreeNode* root) {
        
        inorder(root);
        last_val_node->right = nullptr;
        return ans;
    }
};